### PR TITLE
consensus: remove `round` from sortition hash

### LINKED ISSUE #1204: consensus: `round` in Sortition hash is redundant
#### Summary

The `round` number is not necessary for the uniqueness of the Sortition hash at each round, since this is already ensured by the `seed` (which depends on the previous block).

#### Detailed Description

In the Sortition hash, the `round` field is redundant because the `seed` is already unique for each block, hence ensuring the uniqueness of the hash at each round.

#### Relevant Context

The Sortition hash is intended to be a unique pseudo-random value that is used as a seed for the extraction procedure.
This hash has to be unique for each extraction, so it must vary depending for each Round, Iteration, and Step.
Additionally, it must: (1) depend on the previous block so it can't be predicted, (2) change at every extraction to be fully randomic.

#### Possible Solution

Remove the `round` field from `create_sortition_hash`

### PR DESCRIPTION
**Pull request by herr-seppia**
Resolves #1204

### CODE REVIEW

**Review by fed-franz — APPROVED** [review]
LGTM

File: consensus/src/user/sortition.rs [hunk 1]
```diff
@@ -74,7 +74,6 @@ pub fn create_sortition_hash(cfg: &Config, counter: u32) -> [u8; 32] {
 
     // write input message
     hasher.update(&cfg.seed.inner()[..]);
-    hasher.update(cfg.round.to_le_bytes());
```
Comment (goshawk-3) [inline consensus/src/user/sortition.rs:77]: Shall we remove `round` field from config?

**Review by goshawk-3 — CHANGES_REQUESTED** [review]

**Review by fed-franz — APPROVED** [review]

File: consensus/src/user/sortition.rs (hunk 1 above)
Comment (herr-seppia) [reply consensus/src/user/sortition.rs:77]: No, because it's needed to know the elibigle provisioners

Comment (herr-seppia) [reply consensus/src/user/sortition.rs:77]: Thanks, I'll see if I can remove it next pr

**Review by goshawk-3 — APPROVED** [review]

**Pull request merged**